### ISSUE TITLE
Only absolute URLs are supported

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#5424]
**Describe the bug**

Found this error on beta goerli node

```
Apr-23 17:42:24.086[rest]            ^[[34mdebug^[[39m: Res req-bcyh getAttesterDuties - 200
Apr-23 17:42:24.098[rest]            ^[[31merror^[[39m: Req req-bcyf registerValidator error  Only absolute URLs are supported
TypeError: Only absolute URLs are supported
    at getNodeRequestOptions (/usr/src/lodestar/node_modules/node-fetch/lib/index.js:1305:9)
    at /usr/src/lodestar/node_modules/node-fetch/lib/index.js:1410:19
    at new Promise (<anonymous>)
    at HttpClient.fetch (/usr/src/lodestar/node_modules/node-fetch/lib/index.js:1407:9)
    at HttpClient.fetch (/usr/src/lodestar/node_modules/cross-fetch/dist/node-ponyfill.js:10:20)
    at HttpClient.requestWithBody (file:///usr/src/lodestar/packages/api/src/utils/client/httpClient.ts:278:30)
    at HttpClient.requestWithBodyWithRetries (file:///usr/src/lodestar/packages/api/src/utils/client/httpClient.ts:171:19)
    at HttpClient.request (file:///usr/src/lodestar/packages/api/src/utils/client/httpClient.ts:158:17)
    at Object.request [as registerValidator] (file:///usr/src/lodestar/packages/api/src/utils/client/client.ts:75:37)
    at ExecutionBuilderHttp.registerValidator (file:///usr/src/lodestar/packages/beacon-node/src/execution/builder/http.ts:79:36)
Apr-23 17:42:24.098[rest]            ^[[34mdebug^[[39m: Res req-bcyf registerValidator - 500
Apr-23 17:42:24.098[rest]            ^[[34mdebug^[[39m: Req req-bcyi 127.0.0.1 produceBlindedBlock
```

**Expected behavior**

No error

**Comment by nflaig** [comment]
Recently added better URL validation at startup in https://github.com/ChainSafe/lodestar/pull/5884.
https://github.com/ChainSafe/lodestar/blob/9dae25f817648635d39761b57cf2474e4abe7787/packages/api/src/utils/client/httpClient.ts#L129-L134

An issue like this where an error happens at runtime due to an invalid URL should not happen anymore.

See rationale for updating the validation https://github.com/ChainSafe/lodestar/pull/5884#discussion_r1297428987
> As another side note, if we want to throw an error, this can not be implemented here as it would lazily throw the first time a request is done which is not ideal. We have to throw misconfiguration errors on startup. This is why I also added URL validation in the constructor of the http client, these runtime errors are bad, especially if you use fallback URLs which might not be queried for hours / days and when they do you notice that the URL is invalid.

**Comment by philknows** [comment]
- fixed in #5884
